src/mock/parsec/mod.rs

```diff
@@ -188,6 +188,10 @@ where
     }
 
     pub fn poll(&mut self) -> Option<Block<T, S::PublicId>> {
+        if self.first_unpolled >= self.first_unconsensused {
```

Comment:
This was actually a bug which was never exposed, because we never called `poll` more than once in a row without handling gossip in between. But now we sometimes do that (because `parsec_poll` is now called after every network_event or action). This is a simple fix that makes sure things are sane. That is, we only return a block from `poll` if we previously reached consensus on that block.

Reply:
Ah, right.
maybe better put your explaination as a comment?

Reply:
Will do.